### PR TITLE
Feature/iter cunone

### LINKED ISSUE #2333: [RFC]: add `@stdlib/iter/cunone`
### Description

This RFC proposes adding the package `@stdlib/iter/cunone`, which cumulatively tests whether every iterated value is falsy. The function should be a transform iterator, continuing to iterate while source iterator values are available.

```javascript
var array2iterator = require( '@stdlib/array/to-iterator' );

var arr = array2iterator( [ false, false, false, true, false ] );

var it = iterCuNone( arr );

var v = it.next().value;
// returns true

v = it.next().value;
// returns true

v = it.next().value;
// returns true

v = it.next().value;
// returns false

v = it.next().value;
// returns false

var bool = it.next().done;
// returns true
```

### Related Issues

No.

### Questions

No.

### Other

- See also `@stdlib/iter/none`

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by ruthwikchikoti**
Resolves #2333.

This pull request:
- Implements a new iterator function `iterCuNone`
- Adds functionality to cumulatively test whether every iterated value is falsy
- Enhances the stdlib iterator ecosystem

This pull request:
- resolves #2333

This implementation:
- Takes an input iterator
- Returns a new iterator that yields boolean values
- Continues yielding `true` while all values are falsy
- Yields `false` once a truthy value is encountered
- Handles both iterable and non-iterable input iterators

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/iter/cunone/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var iterCuNone = require( '@stdlib/iter/cunone' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cunone/README.md:40]: This file is incomplete.

File: lib/node_modules/@stdlib/iter/cunone/benchmark/benchmark.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            b.fail( 'should return a boolean' );
+        }
+    }
+    b.toc();
+    if ( !isBoolean( v ) ) {
+        b.fail( 'should return a boolean' );
+    }
+    b.pass( 'benchmark finished' );
+    b.end();
+});
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cunone/benchmark/benchmark.js:111]: Make sure you have EditorConfig setup as documented in the dev guides.

File: lib/node_modules/@stdlib/iter/cunone/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{{alias}}( iterator )
+    Returns an iterator which cumulatively tests whether every iterated value is
+    falsy.
+
+    The returned iterator immediately returns `false` upon encountering a truthy
+    value, for all subsequent iterations.
+
+    If provided an iterator which does not return any iterated values, the
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cunone/docs/repl.txt:8]: This isn't correct.

File: lib/node_modules/@stdlib/iter/cunone/lib/main.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+				'done': true
+			};
+		}
+		v = iterator.next();
+		if ( v.done ) {
+			FLG = true;
+			return v;
+		}
+		if ( v.value ) {
+			FLG = true;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cunone/lib/main.js:106]: I don't think this is what we want. Once this is set, the iterator ends. We want the iterator to continue iterating so long as a source iterator has values.

Comment (kgryte) [reply lib/node_modules/@stdlib/iter/cunone/lib/main.js:106]: Again, this is a cumulative iterator so the iterator should have the same length as the source iterator.

File: lib/node_modules/@stdlib/iter/cunone/package.json [hunk 5]
```diff
@@ -0,0 +1,85 @@
+{
+    "name": "@stdlib/iter-cunone",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cunone/package.json:2]: Package name is incorrect.

File: lib/node_modules/@stdlib/iter/cunone/package.json [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    "homepage": "https://stdlib.io",
+    "repository": {
+      "type": "git",
+      "url": "git://github.com/stdlib-js/iter-cunone.git"
+    },
+    "bugs": {
+      "url": "https://github.com/stdlib-js/stdlib/issues"
+    },
+    "dependencies": {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cunone/package.json:39]: Dependencies and dev deps should not be present.

File: lib/node_modules/@stdlib/iter/cunone/lib/main.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var setReadOnly = require( '@stdlib/utils-define-nonenumerable-read-only-property' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cunone/lib/main.js:23]: All these package names are incorrect. They should be names from the main project, not standalone pkg names. See other packages.

Comment (kgryte) [reply lib/node_modules/@stdlib/iter/cunone/lib/main.js:23]: This applies here and elsewhere.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/README.md [hunk 8]
```diff
@@ -111,6 +111,76 @@ var objectKeys = require( '@stdlib/utils/keys' );
 var kumaraswamy = require( '@stdlib/stats/base/dists/kumaraswamy' );
 
 console.log( objectKeys( kumaraswamy ) );
+// Create a Kumaraswamy distribution object
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/README.md:114]: These changes should not be present in this PR.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/examples/index.js [hunk 9]
```diff
@@ -22,3 +22,72 @@ var objectKeys = require( '@stdlib/utils/keys' );
 var kumaraswamy = require( './../lib' );
 
 console.log( objectKeys( kumaraswamy ) );
+// Create a Kumaraswamy distribution object
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/examples/index.js:25]: These changes should not be present in this PR.

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/examples/index.js (hunk 9 above)
Comment (ruthwikchikoti) [reply lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/examples/index.js:25]: Hey @kgryte, could you clarify if #1632 should be included here? Also, please review #1632 before I move on to #2613.

Comment (ruthwikchikoti) [reply lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/examples/index.js:25]: you are saying to remove #1632 right ? from this issue #2613

Comment (kgryte) [reply lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/examples/index.js:25]: Correct. A PR should be focused on one thing. You are currently mixing both #1632 and #2613. This PR should only be concerned with #2613.

Comment (kgryte) [reply lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/examples/index.js:25]: You can simply undo the Kumaraswamy changes in this PR, and this PR can be judged on its own merits, irrespective of #1632.

**Comment by kgryte** [comment]
@ruthwikchikoti Did you mean to close this? If you need to make changes, just do so on this PR.

**Comment by ruthwikchikoti** [comment]
@kgryte raising a new pr

File: lib/node_modules/@stdlib/iter/cunone/package.json (hunk 5 above)
Comment (ruthwikchikoti) [reply lib/node_modules/@stdlib/iter/cunone/package.json:2]: @stdlib/iter/cunone
@kgryte

Comment (kgryte) [reply lib/node_modules/@stdlib/iter/cunone/package.json:2]: Yes, that is correct. You can confirm this by examining all other packages in the main project repository. In short, package names should match the directory path.

**Pull request closed without merging**